### ISSUE TITLE
Incomplete Testclass initialization

### ISSUE THREAD

**Issue by dasAnderl** [cbeust/testng#1888]
### TestNG Version
TestNg: 6.14.3 
Jmockit 1.40

Problem is that as soon as a setup method throws an exception and jmockit is on the classpath a npe in jmockit is thrown which marks a dependant test method as failure:

```
@BeforeClass
public void setUp() {
	throw new RunTimeException("");
}

@Test
public void test() {}
```

without jmockit on the classpath the test result is as expected:

-setup marked as failure
-test marked as skipped

with jmockit on the classpath the test result is the following.

-setup marked as failure
-test marked as skipped
-test (1) marked as failure

this is due to an npe in jmockit because the testclass object provided by testNg is not properly initialized.

i know this sounds like a jmockit bug, but its actually probably not. i submitted the issue to jmockit and they even identified the part of testNg code which seems to be faulty.

please see https://github.com/jmockit/jmockit1/issues/536#issuecomment-409784913

This problem is reproducable on any runner

**Comment by krmahadevan** [comment]
@dasAnderl 

The sample that was shared in the jmockit comment 

```java
import org.testng.IClass;
import org.testng.IInvokedMethod;
import org.testng.IInvokedMethodListener;
import org.testng.ITestResult;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Listeners;
import org.testng.annotations.Test;

@Listeners(SampleTest.class)
public class SampleTest implements IInvokedMethodListener {

  @Override
  public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
    IClass testClass = testResult.getTestClass(); // this should never be null...
    testClass.getRealClass(); // ... but it is after setUp(), resulting in an NPE
  }

  @Override
  public void afterInvocation(IInvokedMethod method, ITestResult testResult) {
  }

  @BeforeClass // also happens with @BeforeMethod and @BeforeTest
  public void setUp() {
    throw new RuntimeException();
  }

  @Test
  public void anyTest() {
  }
}
```

fails with a `NullPointerException` as you already have figured out

```
java.lang.NullPointerException
	at com.rationaleemotions.github.issue1888.SampleTest.beforeInvocation(SampleTest.java:17)
	at org.testng.internal.invokers.InvokedMethodListenerInvoker$InvokeBeforeInvocationWithoutContextStrategy.callMethod(InvokedMethodListenerInvoker.java:84)
	at org.testng.internal.invokers.InvokedMethodListenerInvoker.invokeListener(InvokedMethodListenerInvoker.java:62)
	at org.testng.internal.Invoker.runInvokedMethodListeners(Invoker.java:493)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:533)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)
```

But the same test when run using the latest snapshot version of TestNG `7.0.0-SNAPSHOT` (this is the upcoming release candidate), the test doesn't show that exception but it just has the `RuntimeException` related details.

```

java.lang.RuntimeException
	at rationale.emotions.issue1888.SampleTest.setUp(SampleTest.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:131)
	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:61)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:518)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:242)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:144)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:173)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:118)
	at org.testng.TestRunner.privateRun(TestRunner.java:724)
	at org.testng.TestRunner.run(TestRunner.java:574)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:398)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:392)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:354)
	at org.testng.SuiteRunner.run(SuiteRunner.java:302)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1142)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1064)
	at org.testng.TestNG.runSuites(TestNG.java:994)
	at org.testng.TestNG.run(TestNG.java:962)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)


Test ignored.
```

So I would suggest that you quickly retry this issue using TestNG `7.0.0-SNAPSHOT` and see if that fixes your problem. If you still have the same problem, we can perhaps debug more.

**Comment by dasAnderl** [comment]
great...thank you for your very quick reply, indeed in 7.0.0-SNAPSHOT the error is gone

**Comment by krmahadevan** [comment]
Based on the above comments, closing this issue with resolution as  **Resolved in 7.0.0**
